Modix.Services/GuildInfo/GuildInfoService.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Discord;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Modix.Services.GuildInfo
+{
+    public class GuildInfoService
+    {
+        private IMemoryCache _cache;
+
+        private readonly MemoryCacheEntryOptions _cacheEntryOptions = 
+            new MemoryCacheEntryOptions().SetSlidingExpiration(TimeSpan.FromDays(1));
```

Comment:
one day might be too long if we want this number to be more or less accurate. Maybe use the guildjoin/guildleave, and role changed events to update this when it changes?

That approach also lets us move the work from this command to something executed when the bot connects initially